docs/understanding-airbyte/jobs.md

```diff
@@ -1,45 +1,111 @@
 
 ![Worker Lifecycle](../.gitbook/assets/worker-lifecycle.png)
 
 [Image Source](https://docs.google.com/drawings/d/1k4v_m2M5o2UUoNlYM7mwtZicRkQgoGLgb3eTOVH8QFo/edit)
 
 See the [architecture overview](high-level-view.md) for more information about workers.
 
-## Worker parallelization
+## Deployment Types
+
+Up to now, the term 'processes' has been used loosely. This section will describe this in more detail.
+
+Airbyte offers two deployment types. The underlying process implementations differ accordingly.
+
+1. The Docker deployment - Each process is a local process backed by a Docker container. As all processes are local, process communication is per standard unix pipes.
+2. The Kubernetes deployment - Each process is a backed by a Kubernetes pod. As Kubernetes does not make process-locality guarantees, Airbyte has implemented mechanisms to hide the remote process execution.
+   See [this blogpost](https://airbyte.com/blog/scaling-data-pipelines-kubernetes) for more details.
+
+### Decoupling Worker and Job Processes
+
+Workers being responsible for all non-connector-related job operations means multiple jobs are operationally dependent on a single worker process.
+
+There are two downsides to this:
+1. Any issues to the parent worker process affects all job processes launched by the worker.
+2. Unnecessary complexity of vertically scaling the worker process to deal with IO and processing requirements from multiple jobs.
+
+This gives us a potentially brittle system component that can be operationally tricky to manage. For example, since redeploying Airbyte terminates all worker processes, all running jobs are also terminated.
+
+The Container Orchestrator was introduced to solve this.
+
+#### Container Orchestrator
+When enabled, workers launch the Container Orchestrator process.
+
+This orchestrator process replaces the previous worker process. It now performs the [above listed responsibilities](#worker-responsibilities).
+
+This decoupling introduces a new need for workers to track the orchestrator's, and the job's, state. This is done via a shared Cloud Storage store.
+
+Brief description of how this works,
+1. Workers constantly poll the Cloud Storage location for job state.
+2. As an Orchestrator process executes, it writes status marker files to the Cloud Storage location i.e. `NOT_STARTED`, `INITIALIZING`, `RUNNING`, `SUCCESS`, `FAILURE`.
+3. If the Orchestrator process runs into issues at any point, it writes a `FAILURE`.
+4. If the Orchestrator process succeeds, it writes a job summary as part of the `SUCCESS` marker file.
+
+The Cloud Storage store is treated as the source-of-truth of execution state.
+
+The Container Orchestrator is only available for Airbyte Kubernetes today and automatically enabled when running the Airbyte Helm charts/Kustomize deploys.
+
```

Comment:
@gosusnp we can add a diagram here in a follow up PR.

Reply:
https://github.com/airbytehq/airbyte/pull/16675